src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,302 @@
+﻿using Markdig.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using static Microsoft.DotNet.Interactive.CSharpProject.RoslynWorkspaceUtilities.BuildCacheFileUtilities;
+
+namespace Microsoft.DotNet.Interactive.CSharpProject.RoslynWorkspaceUtilities;
+
+internal static class RoslynWorkspaceUtilities
+{
+    internal static BuildDataResults GetResultsFromCacheFile(string cacheFilePath)
+    {
+        CodeAnalysis.Workspace ws = null;
+        BuildProjectData buildProjectData = null;
+        CSharpParseOptions cSharpParseOptions = null;
+
+        try
+        {
+            if (!File.Exists(cacheFilePath))
+            {
+                return null;
+            }
+
+            string fileContent = File.ReadAllText(cacheFilePath);
+
+            PopulateBuildProjectData(fileContent, out buildProjectData);
+
+            ws = GetWorkspace(buildProjectData, true);
+
+            cSharpParseOptions = CreateCSharpParseOptions(buildProjectData);
+
+            return new BuildDataResults()
+            {
+                BuildProjectData = buildProjectData,
+                ProjectFilePath = cacheFilePath,
+                CacheFilePath = cacheFilePath + cacheFilenameSuffix,
+                Workspace = ws,
+                Succeeded = true,
+                CSharpParseOptions = cSharpParseOptions
+            };
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    internal static BuildDataResults ResultsFromCacheFileUsingProjectFilePath(string csprojFilePath)
+    {
+        if (!File.Exists(csprojFilePath))
+        {
+            throw new ArgumentException($"project file does not exist : {csprojFilePath}");
+        }
+
+        var cacheFilePath = csprojFilePath + cacheFilenameSuffix;
+
+        return GetResultsFromCacheFile(cacheFilePath);
+    }
+
+    internal static void PopulateBuildProjectData(string fileContent, out BuildProjectData buildProjectData)
+    {
+        buildProjectData = new BuildProjectData();
+
+        string[] lines = fileContent.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries);
+
+        var referencesList = new List<string>();
+        var projectReferencesList = new List<string>();
+        var analyzerReferencesList = new List<string>();
+        var preprocessorSymbolsList = new List<string>();
+        var sourceFilesList = new List<string>();
+
+        foreach (string line in lines)
+        {
+            string[] parts = line.Split('=');
+            if (parts.Length == 2)
```

Comment:
What if the value contains an `=`?

Reply:
This parser is reading values from the auto-generated cache file. This cache file contains values from msbuild properties, and if one of this values are corrupted, then the build will fail.